libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/ClientManager.java

```diff
@@ -790,6 +801,19 @@ public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUr
             this.setJwt(jwt);
         }
 
+        public LoginOptions(String loginUrl, String passcodeHash, String oauthCallbackUrl,
+                            String oauthClientId, String[] oauthScopes, String clientSecret, String jwt,
+                            HashMap<String,String> additionalParameters) {
+            this(loginUrl, passcodeHash, oauthCallbackUrl, oauthClientId, oauthScopes,clientSecret,jwt);
+            this.setAdditionalParameters(additionalParameters);
+        }
+
+        public void setAdditionalParameters(HashMap<String,String> additionalParameters) {
+            this.additionalParameters = additionalParameters;
+            bundle.putSerializable(AuthenticatorService.KEY_ADDL_PARAMS,additionalParameters);
```

Comment:
If this bundle addition is required, should we not expose 'additionalParameters' publicly, so that we make sure all writes to it come through this method?